File: 프로그래머스/K진수에서_소수개수구하기.java
import java.util.*;
class Solution {
    public int solution(int n, int k) {
        int answer = 0;
        String numWord = K진수구하기(n,k,"");
        for(String w : numWord.split("0")){
            if(!w.isEmpty() && 소수구하기(Long.parseLong(w))) answer++;
        }
        return answer;
    }

    static String K진수구하기(int n, int k,String temp){
        if(n == 0) {
            return temp;
        }
        return K진수구하기(n / k, k, (n % k) + temp);
    }
    static String 문자열뒤집기(String word){
        StringBuilder sb = new StringBuilder();
        for(int i=word.length()-1;i >= 0;i--){
            sb.append(word.charAt(i));
        }
        return sb.toString();
    }
    static boolean 소수구하기(long num){
        if(num < 2) return false;
        if(num == 2) return true;
        if(num % 2 == 0) return false;
        for(int i=3; i<=Math.sqrt(num); i+=2){
            if(num % i == 0) return false;
        }
        return true;
    }
}

File: 프로그래머스/K진수에서_소수개수_구하기.js
function solution(n, k) {
    var answer = 0;
    var nums = parse(n,k).split('0');
    for(let i of nums){
        let currentNum = parseInt(i);
        if(!isNaN(currentNum)){
            if(isPrime(currentNum)){
                 answer++;
            }
        }
    }
    return answer;
}

const parse = (n,k) => {
    let number  = '';
    while(n > 0){
        number += String(n % k);
        n = Math.floor(n / k);
    }
    return number.split('').reverse().join('');
}

const isPrime = (n) => {
    if(n < 2) return false;
    for(let i=2;i<=Math.sqrt(n);i++){
        if(n % i ==0) return false;
    }
    return true;
}
